### PR TITLE
Added routing for entities

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | TODO

#### What's in this PR?

This PR add the routing component to create routes for saved entities.

#### Why?

You should be able to browse created routes.

#### To Do

- [ ] Create a documentation PR

### CODE REVIEW

File: DependencyInjection/CompilerPass/DefaultsProviderCompilerPass.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+{
+    const SERVICE_ID = 'sulu_route.routing.defaults_provider';
+    const TAG_NAME = 'sulu_route.defaults_provider';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->getParameter('sulu.context') !== SuluKernel::CONTEXT_WEBSITE) {
```
Comment (chirimoya) [inline DependencyInjection/CompilerPass/DefaultsProviderCompilerPass.php]: use yoda conditions

File: Resources/config/routing.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <parameters>
+        <parameter key="sulu_route.routing.uri_filter_regexp">null</parameter>
+    </parameters>
+
+    <services>
+        <!-- defaults -->
```
Comment (chirimoya) [inline Resources/config/routing.xml]: route defaults

**Comment by wachterjohannes** [comment]
@chirimoya the comments are fixed

File: Routing/Defaults/RouteDefaultsProvider.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    public function supports($entityClass)
+    {
+        return null !== $this->getDefaultProvider($entityClass);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function getDefaultProvider($entityClass)
+    {
```
Comment (chirimoya) [inline Routing/Defaults/RouteDefaultsProvider.php]: cache default provider

File: Routing/RouteProvider.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    /**
+     * @var RouteDefaultsProviderInterface
+     */
+    private $routeDefaultsProvider;
+
+    public function __construct(
```
Comment (chirimoya) [inline Routing/RouteProvider.php]: phpdoc

File: Routing/Defaults/RouteDefaultsProvider.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ * Combines multiple defaults-provider.
+ */
+class RouteDefaultsProvider implements RouteDefaultsProviderInterface
+{
+    /**
+     * @var RouteDefaultsProviderInterface[]
+     */
+    private $defaultsProvider;
+
+    public function __construct(array $defaultsProvider)
```
Comment (chirimoya) [inline Routing/Defaults/RouteDefaultsProvider.php]: phpdoc

**Pull request merged**